matrix-sdk-android/src/main/java/org/matrix/android/sdk/internal/session/space/ResolveSpaceInfoTask.kt

```diff
@@ -28,22 +29,39 @@ internal interface ResolveSpaceInfoTask : Task<ResolveSpaceInfoTask.Params, Spac
             val maxDepth: Int?,
             val from: String?,
             val suggestedOnly: Boolean?
-//            val autoJoinOnly: Boolean?
     )
 }
 
 internal class DefaultResolveSpaceInfoTask @Inject constructor(
         private val spaceApi: SpaceApi,
         private val globalErrorReceiver: GlobalErrorReceiver
 ) : ResolveSpaceInfoTask {
-    override suspend fun execute(params: ResolveSpaceInfoTask.Params): SpacesResponse {
-        return executeRequest(globalErrorReceiver) {
+
+    override suspend fun execute(params: ResolveSpaceInfoTask.Params) = executeRequest(globalErrorReceiver) {
+        getSpaceHierarchy(params)
+    }
+
+    private suspend fun getSpaceHierarchy(params: ResolveSpaceInfoTask.Params) = try {
```

Comment:
I found using `= try` and spilling into 2 methods less readable than a simple
```kotlin
override suspend fun execute(params: ResolveSpaceInfoTask.Params) {
    executeRequest(globalErrorReceiver) {
        try {
            getUnstableSpaceHierarchy(params)
        } catch (e: HttpException) {
            getStableSpaceHierarchy(params)
        }
    }
}
```

Also I would rename `getStableSpaceHierarchy` to `getSpaceHierarchy` (rule: never use `stable`, just use `unstable`, intended to be removed after a while)

Reply:
Good shout!